crates/uv-workspace/src/settings.rs

```diff
@@ -0,0 +1,84 @@
+use serde::Deserialize;
+use std::path::PathBuf;
+
+use distribution_types::{FlatIndexLocation, IndexUrl};
+use install_wheel_rs::linker::LinkMode;
+use uv_auth::KeyringProvider;
+use uv_configuration::{ConfigSettings, IndexStrategy, PackageNameSpecifier};
+use uv_normalize::PackageName;
+use uv_resolver::{AnnotationStyle, ExcludeNewer, PreReleaseMode, ResolutionMode};
+use uv_toolchain::PythonVersion;
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct PyProjectToml {
+    pub(crate) tool: Option<Tools>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct Tools {
+    pub(crate) uv: Option<Options>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Options {
+    pub quiet: Option<bool>,
+    pub verbose: Option<bool>,
+    pub native_tls: Option<bool>,
+    pub no_cache: bool,
+    pub cache_dir: Option<PathBuf>,
+    pub pip: Option<PipOptions>,
+    pub resolver: Option<ResolverOptions>,
+    pub installer: Option<InstallerOptions>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct PipOptions {
+    pub system: Option<bool>,
+    pub offline: Option<bool>,
+    pub index_url: Option<IndexUrl>,
+    pub extra_index_url: Option<IndexUrl>,
+    pub no_index: Option<bool>,
+    pub find_links: Option<Vec<FlatIndexLocation>>,
+    pub index_strategy: Option<IndexStrategy>,
+    pub keyring_provider: Option<KeyringProvider>,
+    pub no_build: Option<bool>,
+    pub no_binary: Option<Vec<PackageNameSpecifier>>,
+    pub only_binary: Option<Vec<PackageNameSpecifier>>,
+    pub no_build_isolation: Option<bool>,
+    pub resolver: Option<ResolverOptions>,
+    pub installer: Option<InstallerOptions>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct ResolverOptions {
+    pub resolution: Option<ResolutionMode>,
+    pub prerelease: Option<PreReleaseMode>,
+    pub no_strip_extras: Option<bool>,
+    pub no_annotate: Option<bool>,
+    pub no_header: Option<bool>,
```

Comment:
Aren't a bunch of these very pip-compile output specific? I find it weird to consider these resolver options.

Reply:
Yes, absolutely. Hence: https://github.com/astral-sh/uv/pull/3007#discussion_r1566441095